telegrinder/tools/global_context/global_context.py

```diff
@@ -0,0 +1,269 @@
+
+    @root_protection
+    def __setattr__(self, __name: str, __value: CtxValue):
+        """Setting a root attribute or context variable."""
+        if self.is_root_attribute(__name):
+            return object.__setattr__(self, __name, __value)
+        ctx_storage = self.__get_context_storage()
+        var = ctx_storage.get(__name)
+        if var is not None and var.const:
+            raise TypeError(
+                f"Unable to set variable {__name!r}, because it's a constant."
+            )
+        ctx_storage[__name] = GlobalCtxVar.collect(__name, __value)
+
+    @root_protection
+    def __getattr__(self, __name: str) -> typing.Any:
+        """Getting a root attribute or context variable."""
+        if self.is_root_attribute(__name):
+            return object.__getattribute__(self, __name)
+        return self.get(__name).unwrap().value
+
+    @root_protection
+    def __delattr__(self, __name: str):
+        """Removing a context variable."""
+        var = self.get(__name).unwrap()
+        if var.const:
+            raise TypeError(
+                f"Unable to delete variable {__name!r}, because it's a constant."
+            )
+        del self.__get_context_storage()[__name]
+
+    def __get_context_storage(self) -> dict[str, GlobalCtxVar[typing.Any]]:
+        """Returns context storage by `self.__ctx_name__`
+        If `self.__ctx_name__` is None returns main context storage
+        or a context that refers to the specified name
+        if it does not exist it is created and returned.
+        """
+        main_ctx_storage = self.__class__.__context_storage
+        other_ctxs = main_ctx_storage["__other_ctxs__"].value
+        if self.__ctx_name__ is None:
+            return main_ctx_storage
+        return other_ctxs.setdefault(self.__ctx_name__, dict())
+
+    def __copy_context_storage(
+        self,
+        *,
+        exclude_root_context_vars: bool = False,
+    ) -> dict[str, GlobalCtxVar[typing.Any]]:
+        ctx = self.__get_context_storage().copy()
+        if exclude_root_context_vars:
+            for root_ctx_var in self.__root_context_variables__:
+                ctx.pop(root_ctx_var, None)
+        return ctx
+
+    def is_root_context_variable(self, name: str) -> bool:
+        """Returns True if exists root context variable
+        otherwise False."""
+        return name in self.__root_context_variables__
+
+    def is_root_attribute(self, name: str) -> bool:
+        """Returns True if exists root attribute
+        otherwise False."""
+        return name in self.__root_attributes__
+
+    def get_root_attribute(self, name: str) -> Result[RootAttr, str]:
+        """Get root attribute by name."""
+        if self.is_root_attribute(name):
+            for rattr in self.__root_attributes__:
+                if rattr.name == name:
+                    return Ok(rattr)
+        return Error(f"Root attribute {name!r} not found.")
+
+    def get(
```

Comment:
Perhaps its better to return actual value and add typecheck for value_type

Reply:
It is possible to realize obtaining a value with checking of the specified data type for 'get_value' method. I'll leave the 'get' method it was created to get a context variable.